ctest-next/templates/test.c

```diff
@@ -34,3 +34,21 @@ static {{ constant.c_ty }} ctest_const_{{ constant.id }}_val_static = {{ constan
     return &ctest_const_{{ constant.id }}_val_static;
 }
 {%- endfor +%}
+
+{%- for item in ctx.size_align_tests +%}
+
+// Return the size of a type.
+uint64_t ctest_size_of__{{ item.id }}(void) { return sizeof({{ item.c_ty }}); }
+
+// Return the alignment of a type.
+uint64_t ctest_align_of__{{ item.id }}(void) { return _Alignof({{ item.c_ty }}); }
+{%- endfor +%}
+
+{%- for alias in ctx.signededness_tests +%}
+
+// Return `1` if the type is signed, otherwise return `0`.
+// Casting -1 to the aliased type if signed evaluates to `-1 < 0`, if unsigned to `MAX_VALUE < 0`
+uint32_t ctest_signededness_of__{{ alias.id }}(void) {
+    return ((({{ alias.c_ty }}) -1) < 0);
```

Comment:
I think this is a bit easier to read

```suggestion
    { alias.c_ty } all_ones = -1;
    return all_ones < 0;
```